app/code/Magento/MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/Keyword.php

```diff
@@ -37,14 +37,19 @@ public function __construct(ResourceConnection $resource)
     public function apply(Filter $filter, AbstractDb $collection): bool
     {
         $value = $filter->getValue();
+        if (!is_array($value)) {
+            $keywordValues = explode(',', $value);
```

Comment:
I believe the keyword value should bee processed in `getAssetIdsByKeyword` method.

Right now the where condition for keywords is `keyword = ?`, it could be `keyword in (?)`